File: interviewprep/c++/arrayInsert.cpp
#include<iostream>
using namespace std;

void insert(int a[],int n,int pos,int size){
	for(int i=size-1;i>=pos;i--){
		a[i+1] = a[i];
	}
	a[pos] = n;
}
void show(int a[],int n){
	for(int i=0;i<n;i++){
		cout<<a[i]<<"\t";
	}
}
int main(){
int a[] = {12,13,15,16,17};
int size = sizeof(a)/sizeof(a[0]);
cout<<"The input array is: \n";
show(a,size);
int n = 14;
int pos = 2;
cout<<"\nElement to be inserted is: "<<n<<" at position: "<<pos;
cout<<"\nArray after the insertion operation:\n";
insert(a,n,pos,size);
size++;
show(a,size);
cin.get();
return 0;
}

File: interviewprep/java-revision/arrayInsert.java

public class arrayInsert{
	public static void insert(int[] a,int[] p,int n,int pos){
		for(int i=a.length-1;i>=pos;i--){
			p[i+1] = a[i];
		}
		p[pos] = n;
		for(int j=0;j<pos;j++){
			p[j] = a[j];
		}
	}
	public static void main(String[] args){
		int[] a = {12,13,15,16,17};
		int i;
		for(i=0;i<a.length;i++){
			System.out.print(a[i]+"\t");
		}
		System.out.println();
		int n=14;
		int pos=2;
		System.out.println("Element to be added: "+n+" at position: "+pos);
		int[] p =new int[a.length+1];		
		insert(a,p,n,pos);
		for(i=0;i<p.length;i++){
			System.out.print(p[i]+"\t");
		}
	}
}
